src/components/RefinementList/RefinementList.tsx

```diff
@@ -130,44 +223,43 @@ class RefinementList extends Component {
         return;
       }
 
-      if (parent.tagName === 'A' && parent.href) {
+      if (parent.tagName === 'A' && (parent as HTMLAnchorElement).href) {
         originalEvent.preventDefault();
       }
 
-      parent = parent.parentNode;
+      parent = parent.parentNode as HTMLElement;
     }
 
     originalEvent.stopPropagation();
 
-    this.refine(facetValueToRefine, isRefined);
+    this.refine(facetValueToRefine);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.searchBox && !nextProps.isFromSearch) {
-      this.searchBox.resetInput();
+  public componentWillReceiveProps(
+    nextProps: RefinementListPropsWithDefaultProps<TTemplates, TCSSClasses>
+  ) {
+    if (this.searchBox.current && !nextProps.isFromSearch) {
+      this.searchBox.current.resetInput();
     }
   }
 
-  refineFirstValue() {
-    const firstValue = this.props.facetValues[0];
+  private refineFirstValue() {
+    const firstValue = this.props.facetValues && this.props.facetValues[0];
     if (firstValue) {
       const actualValue = firstValue.value;
       this.props.toggleRefinement(actualValue);
     }
   }
 
-  render() {
-    // Adding `-lvl0` classes
-    const cssClassList = cx(this.props.cssClasses.list, {
-      [`${this.props.cssClasses.depth}${this.props.depth}`]: this.props
-        .cssClasses.depth,
-    });
```

Comment:
`cssClasses.depth` is never set by the widget, so if I'm right this code always default to `cssClasses.list`.

(I also removed the related test, that was working since we were passing the `depth` class directly without using the widget)

Reply:
also not with hierarchical menu?

Reply:
Nothing in hierarchical menu either https://github.com/algolia/instantsearch.js/blob/8e3c406c8f29bcae56d2f82f07cbd087043346fe/src/widgets/hierarchical-menu/hierarchical-menu.js#L197-L225